src/index.ts

```diff
@@ -2,10 +2,19 @@ import { base58btc, blake2b256, type Digest } from '@chelonia/multiformats'
 import scrypt from 'scrypt-async'
 import nacl from 'tweetnacl'
 
-const strToBuf = (str: string) => {
+const bufToStr = (() => {
+  const textDecoder = new TextDecoder()
+  return (str: Uint8Array) => {
```

Comment:
Question: would it make sense to rename `str` here to `buf`?

And perhaps give return-value type annotations to both `bufToStr` and `strToBuf` functions?

Reply:
>  rename str here to buf?

That'd indeed make sense

> perhaps give return-value type annotations

That's not necessary since they can be inferred. For example, `strToBuf` already has the type `(buf: string) => Uint8Array` (inferred from the `encode`)